concent_api/core/tests/test_core_views.py

```diff
@@ -129,6 +129,41 @@ def test_send_should_return_http_200_if_message_timeout(self):
         )
         self.assertIsInstance(response_message, GolemMessage)
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_accept_valid_message_with_non_numeric_task_id(self):
```

Comment:
Could you extend this test to have a list of valid and invalid ids and check them in a loop?

Reply:
What would be invalid id ?

Reply:
So to Golem there are no invalid IDs... that's bad because I think that using stuff like `[]`, `{}` `None`, `b'abc'`, `(1, 2, 3)`, `ForceReportComputedTask()` as IDs should not be allowed. It can easily lead to unexpected behavior if you don't carefully think about every possible corner case.

I think that you should at least add a test that checks these values. And you should test both `send/` and `receive/`.

Reply:
I will add them as soon as we clarify with Golem team to use only strings.